src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -34,13 +50,391 @@ TriangleCountExecutor::TriangleCountExecutor(SQLiteDBInterface *db, PerformanceS
+        sqlite->runSelect("SELECT DISTINCT idworker,ip,server_port FROM worker;");
+    for (int i = 0; i < results.size(); i++) {
+        string workerId = results[i][0].second;
+        string ip = results[i][1].second;
+        string port = results[i][2].second;
+        workers[workerId] = ip + ":" + port;
+    }
+
+    map<string, int> loads;
+    const map<string, string> &cpu_map = Utils::getMetricMap("cpu_usage");
+    for (auto it = workers.begin(); it != workers.end(); it++) {
+        auto workerId = it->first;
+        auto worker = it->second;
+        const auto workerLoadIt = cpu_map.find(worker);
+        if (workerLoadIt != cpu_map.end()) {
+            double load = stod(workerLoadIt->second.c_str());
+            if (load < 0) load = 0;
+            loads[workerId] = (int)(4 * load);
+        } else {
+            loads[workerId] = 0;
+        }
+    }
+
+    std::map<int, std::vector<string>> p_avail;
+    std::set<int> remain;
+    for (auto it = partitionMap.begin(); it != partitionMap.end(); it++) {
+        auto worker = it->first;
+        auto &partitions = it->second;
+        for (auto partitionIt = partitions.begin(); partitionIt != partitions.end(); partitionIt++) {
+            auto partition = stoi(*partitionIt);
+            p_avail[partition].push_back(worker);
+            remain.insert(partition);
+        }
+    }
+    const std::map<int, std::vector<string>> P_AVAIL = p_avail;  // get a copy and make it const
+
+    for (auto loadIt = loads.begin(); loadIt != loads.end(); loadIt++) {
+        if (loadIt->second < 3) continue;
+        auto w = loadIt->first;
+        for (auto it = p_avail.begin(); it != p_avail.end(); it++) {
+            auto &partitionWorkers = it->second;
+            for (auto workerIt = partitionWorkers.begin(); workerIt != partitionWorkers.end();) {
+                if (*workerIt == w) {
+                    partitionWorkers.erase(workerIt);
+                } else {
+                    workerIt++;
+                }
+            }
+        }
+    }
+
+    std::map<int, string> alloc;
+    int unallocated = alloc_plan(alloc, remain, p_avail, loads);
+    if (unallocated > 0) {
+        triangleCount_logger.info(to_string(unallocated) + " partitions remaining after alloc_plan");
+        auto copying = reallocate_parts(alloc, remain, P_AVAIL);
+        scale_up(loads, workers, copying.size());
+        triangleCount_logger.info("Scale up completed");
+
+        map<string, int> net_loads;
+        for (auto it = loads.begin(); it != loads.end(); it++) {
+            net_loads[it->first] = 0;
+        }
+        for (auto it = workers.begin(); it != workers.end(); it++) {
+            if (loads.find(it->first) == loads.end()) {
+                loads[it->first] = 3;
+            }
+        }
+
+        std::map<int, std::pair<string, string>> transfer;
+        int net_load = alloc_net_plan(alloc, copying, transfer, net_loads, loads, P_AVAIL, 100000000);
```

Comment:
Rather than having numeric literal 100000000, lets assign it to a constant and refer it here.